modelzoo/admin.py

```diff
@@ -0,0 +1,475 @@
+import grpc
+import typing
+from uuid import UUID
+from _protos.services_pb2_grpc import ModelStub
+from _protos.services_pb2 import (
+    AuthenticationRequest,
+    RegisterTokenRequest,
+    RegisterUserRequest,
+    AuthenticationResponse,
+    TextGenerationRequest,
+    ImageSegmentationRequest,
+    VisionClassificationRequest,
+    ModelUUIDRequest,
+    ModelUUIDResponse,
+    ModelResponse,
+)
+import utils
+
+ModelOutput = typing.NewType(
+    "ModelOutput",
+    typing.Union[typing.Sequence[str], type(Image), typing.Sequence[dict]],
+)
+
+
+def validate_uuid(token: str) -> bool:
+    try:
+        token_id = UUID(token, version=4)
+    except ValueError:
+        return False
+    return True
+
+
+class UserAuth(object):
+    def __init__(
+        self,
+        conn: ModelZooConnection,
+        username: typing.Optional[str] = "",
+        password: typing.Optional[str] = "",
+        token: typing.Optional[str] = "",
+        admin: typing.Optional[bool] = True,
+    ):
+        if token == "":
+            if username == "" or password == "":
+                raise InsufficientCredentialsException(
+                    "Username and Password or Token are required to authenticate."
+                )
+        self.username = username
+        self.password = password
+        self.use_token = True
+        if token != "" and not validate_uuid(token):
+            raise InvalidTokenException("Invalid Token.")
+        if token == "":
+            self.use_token = False
+        else:
+            self.token = token
+
+        self.adminF = admin
+        self.admin_token = None
+        if admin:
+            self.admin_token = conn._authenticate(self.username, self.password)
+        self.parent = None
+
+    def make(
```

Comment:
This is a staticmethod? Otherwise the constructor will have all the data, why are `conn, username, ...` still passed in there?

Reply:
This is to make a new admin token. For example, if a user authenticates within a method with a different login.